javascript/js-lowerings/src/org/jetbrains/dukat/js/lowerings/lowerings.kt

```diff
@@ -0,0 +1,145 @@
+
+        for(jsParameter in parameters) {
+            parameterModels.add(jsParameter.convert())
+        }
+
+        return FunctionModel(
+                name = IdentifierEntity(name),
+                parameters = parameterModels,
+                type = ANY_NULLABLE_TYPE,
+                typeParameters = emptyList<TypeParameterModel>(),
+
+                annotations = mutableListOf(MODULE_ANNOTATION),
+
+                export = true,
+                inline = false,
+                operator = false,
+
+                extend = null,
+                body = emptyList<StatementModel>(),
+                visibilityModifier = VisibilityModifier.DEFAULT
+        )
+    }
+
+    private fun JSClassDeclaration.convert(): ClassModel {
+        return ClassModel(
+                name = IdentifierEntity(name),
+                members = listOf<MemberModel>(),
+                companionObject = null,
+                typeParameters = emptyList(),
+                parentEntities = emptyList(),
+                primaryConstructor = null,
+                annotations = mutableListOf(MODULE_ANNOTATION),
+                comment = null,
+                external = true,
+                abstract = false,
+                visibilityModifier = VisibilityModifier.DEFAULT
+        )
+    }
+
+    private fun JSDeclaration.convert(): TopLevelModel {
+        return when(this) {
+            is JSFunctionDeclaration -> this.convert()
+            is JSClassDeclaration -> this.convert()
+            is JSReferenceDeclaration -> throw IllegalStateException("References need to be resolved before conversion to model.")
+            else -> throw IllegalStateException("Declaration of type <${this.javaClass}> cannot be converted.")
+        }
+    }
+
+    private fun resolve(reference: JSReferenceDeclaration): JSDeclaration? {
+        return moduleDeclaration.topLevelDeclarations[reference.name]
+    }
+
+    private fun resolveReferencedExportDeclarations() {
+        for(declaration in moduleDeclaration.exportDeclarations) {
+            if(declaration is JSReferenceDeclaration) {
+                moduleDeclaration.exportDeclarations.remove(declaration)
+
+                var resolvedDeclaration: JSDeclaration? = declaration
+
+                while(resolvedDeclaration is JSReferenceDeclaration) {
+                    resolvedDeclaration = resolve(resolvedDeclaration)
+                }
+
+                if(resolvedDeclaration != null) {
+                    moduleDeclaration.exportDeclarations.add(resolvedDeclaration)
+                } else {
+                    logger.warn("Cannot resolve reference for export: " + declaration.name)
+                }
+            }
+        }
+    }
+
+    fun lower() : SourceSetModel {
+        resolveReferencedExportDeclarations()
+
+        val moduleContents: MutableList<TopLevelModel> = mutableListOf()
+
+        for(exportDeclaration in moduleDeclaration.exportDeclarations) {
```

Comment:
Use just map

Reply:
With only a single export, this no longer exists (but will try to remember using map in the future)